File: mpp/lab7/client/src/main/java/domain/ScoreType.java
package domain;

import grpc.triathlon.ScoreTypeProto;

public enum ScoreType {
    CYCLING,
    RUNNING,
    SWIMMING;

    public static ScoreType fromInteger(int value) {
        switch (value) {
            case 0:
                return CYCLING;
            case 1:
                return RUNNING;
            case 2:
                return SWIMMING;
            default:
                throw new IllegalArgumentException();
        }
    }

    public static ScoreType fromProto(ScoreTypeProto proto) {
        switch (proto) {
            case CYCLING:
                return CYCLING;
            case RUNNING:
                return RUNNING;
            case SWIMMING:
                return SWIMMING;
            case UNRECOGNIZED:
                throw new IllegalArgumentException();
        }

        return null;
    }
}

File: mpp/lab7/server/Domain/ScoreType.cs
using System;
using Triathlon;

namespace server.Domain
{
    public enum ScoreType
    {
        Cycling = 0,
        Running = 1,
        Swimming = 2,
    }

    public static class ScoreTypeUtils
    {
        public static ScoreTypeProto ToProto(ScoreType type)
        {
            return type switch
            {
                ScoreType.Cycling => ScoreTypeProto.Cycling,
                ScoreType.Running => ScoreTypeProto.Running,
                ScoreType.Swimming => ScoreTypeProto.Swimming,
                _ => throw new InvalidCastException()
            };
        }
    }
}
